src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -248,23 +250,27 @@ private Binder GetBinder(SyntaxNode syntax)
 
         private void NullableAnalyzeParameterDefaultValueFromAttributes()
         {
-            if (!NullableWalker.NeedsAnalysis(DeclaringCompilation))
+            var parameterSyntax = this.CSharpSyntaxNode;
+            if (parameterSyntax == null)
             {
+                // If there is no syntax at all for the parameter, it means we are in a situation like
+                // a property setter whose 'value' parameter has a default value from attributes.
+                // There isn't a sensible use for this in the language, so we just bail in such scenarios.
                 return;
             }
 
-            var defaultValue = DefaultValueFromAttributes;
-            if (defaultValue == null || defaultValue.IsBad)
+            // The syntax span used to determine whether the attribute value is in a nullable-enabled
+            // context is larger than necessary - it includes the entire attribute list rather than the specific
+            // default value attribute which is used in AttributeSemanticModel.IsNullableAnalysisEnabled().
+            var attributes = parameterSyntax.AttributeLists.Node;
```

Comment:
What node are we going to get here?

Reply:
The underlying node for the `SyntaxList<AttributeListSyntax>` from the `ParameterSyntax`. 

---
In reply to: [550259550](https://github.com/dotnet/roslyn/pull/49953#discussion_r550259550) [](ancestors = 550259550)